src/iosource/pcap/Source.cc

```diff
@@ -186,6 +186,13 @@ void PcapSource::OpenOffline()
 		InternalError("OS does not support selectable pcap fd");
 
 	props.link_type = pcap_datalink(pd);
+#ifdef DONT_HAVE_LIBPCAP_DLT_LINUX_SLL2
+	if ( props.link_type == 276 )
+		{
+		Error("PCAP file is using DLT_LINUX_SLL2 which is unsupported by your libpcap version");
```

Comment:
I'm a little concerned about reporting an error here sinc reporter messages aren't rate-limited it at all. If you're getting a lot of these packets on an unsupported system your reporter log will explode. I don't think the previous weird of `unknown_ip_version` is correct here though. I wonder if we should just silently return instead, but that doesn't feel right either.

What does libpcap do if it gets one of these and it doesn't support the link type? Is the packet object not well-formed? Could we not just let the processing flow happen and let the SLL2 analyzer check it?

Reply:
I haven't tested, but this is in `PcapSource::OpenOffline()` rather than on a per-packet basis and we skip the `Opened()` callback. Wouldn't we just run into this and exit then?

https://github.com/zeek/zeek/blob/f5df2a132a0cf6e6601404968a9977b50a9685e1/src/RunState.cc#L157-L159

Reply:
@awelzel correct, it just exits while opening the PCAP source. My comment above: https://github.com/zeek/zeek/pull/2340#issuecomment-1219440742 shows the before/after behaviour.

@timwoj interestingly libpcap < 1.10.0 is blissfully ignorant about the link type, and if all you do is this, it works:
```c
char errbuf[PCAP_ERRBUF_SIZE];
struct pcap_pkthdr *header;
const u_char *pkt_data;

fp = pcap_open_offline(filename, errbuf);

while ( pcap_next_ex( fp, &header, &pkt_data) >= 0 ) {
    // do something with the packet
}
``` 

Specifically, `pkt_data` contains the full packet including link layer (which zeek would be able to handle with the new analyzer).

However if you try this:
```c
struct bpf_program program;

int res == pcap_compile(fp, &program, "ip or not ip", 0, PCAP_NETMASK_UNKNOWN);
if (res != 0) {
    printf("Error compiling filter [%d]: %s\n", res, pcap_geterr(fp));
}
```
...you get `Error compiling filter [-2]: unknown data link type 276`, which is exactly what happens in zeek. I thought about conditionally disabling the BPF code but that seems like a case of diminishing returns for the extra code complexity.

Since zeek would exit anyway due to the BPF error, adding this check improves the error message and handles the condition in the most logical place (perhaps preventing future surprises if the BPF code changes).

Reply:
I'm hoping to fix the BPF error in a more generic fashion via https://github.com/zeek/zeek/issues/2132, probably before 5.1 comes out.

Reply:
@timwoj ok, I see. Please let me know if you'd like me to remove this then.

Reply:
I'd say remove it. If it turns out to be a big problem, we can always add it back in. I just don't want to end up special-casing a bunch of link types in `Packet`, if possible.